### PR TITLE
feat: Ability to send attachment in new conversation

### PR DESCRIPTION
**Pull request by iamsivin**
# Pull Request Template

## Description

This PR allows sending attachments while creating a new conversation for the modal.

Fixes https://linear.app/chatwoot/issue/CW-2301/feat-ability-to-send-attachment-in-new-conversation

Reverted PR https://github.com/chatwoot/chatwoot/pull/7698

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

**Test cases**

1. Send messages from the new conversation modal with and without attachments.
2. Send messages through all inboxes except Whatsapp Cloud
3. Try to drag and drop through the conversation and contact pages.
4. Try attaching the through button from the modal, and try multiple attachments (manually and drag).
5. Check the reply box attachments flow (drag and drop)
6. Check drag and drop with the new conversation modal active in the conversation screen.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by sojan-official** [comment]
Will test the functionality and will get this merged.

**Comment by iamsivin** [comment]
> Will test the functionality and will get this merged.

Thanks

**Comment by pranavrajs** [comment]
@muhsin-k @scmmishra Can you please test this out and get this merge?

File: app/javascript/dashboard/components/widgets/conversation/ReplyBox.vue [hunk 1]
```diff
@@ -582,11 +581,25 @@ export default {
       500,
       true
     );
+
+    // A hacky fix to solve the drag and drop
+    // Is showing on top of new conversation modal drag and drop
+    // TODO need to find a better solution
+    bus.$on(
+      BUS_EVENTS.NEW_CONVERSATION_MODAL,
```
Comment (muhsin-k) [inline app/javascript/dashboard/components/widgets/conversation/ReplyBox.vue:589]: @iamsivin Are you emitting this event from anywhere?

Comment (muhsin-k) [reply app/javascript/dashboard/components/widgets/conversation/ReplyBox.vue:589]: The name should be `BUS_EVENTS.NEW_CONVERSATION_MODAL`

**Comment by scmmishra** [comment]
@iamsivin is this ready for review? Also PTAL at the merge conflict

**Comment by iamsivin** [comment]
> @iamsivin is this ready for review? Also PTAL at the merge conflict

@scmmishra Yes it's ready for review

**Comment by scmmishra** [comment]
> > @iamsivin is this ready for review? Also PTAL at the merge conflict
> 
> @scmmishra Yes it's ready for review

Cool, I'll take a look. Also can you add a note on the PR on why the previous commit was reverted and how does this fix it?

**Comment by iamsivin** [comment]
> > > @iamsivin is this ready for review? Also PTAL at the merge conflict
> 
> > 
> 
> > @scmmishra Yes it's ready for review
> 
> 
> 
> Cool, I'll take a look. Also can you add a note on the PR on why the previous commit was reverted and how does this fix it?

Yes, here is the note

The previous commit was reverted because of WhatsApp Cloud's new message failure issues. For supporting attachments, I used form data for sending messages to all inboxes. So, in the WhatsApp cloud, the attachment is not supported, and there is an extra object passing as payload called template_params, which is an object that is not supported through form data. 

So I added a check that, if the message is created from WhatsApp, then it will set the payload the same as before, and for other channels, it's sent through form data.

**Comment by iamsivin** [comment]
> @iamsivin is this ready for review? Also PTAL at the merge conflict

@scmmishra Merge conflicts are resolved. PTAL

**Review by muhsin-k — APPROVED** [review]

**Pull request merged**